### ISSUE TITLE
[Help]: When it comes to interferogram, an error occurs that it reminds me some .grd files are missing.

### ISSUE THREAD

**Issue by Uncleyyyyy** [gmtsar/gmtsar#287]
**Describe the problem you met**
When I'm running a D-InSAR workflow in TOPS mode by the script p2p_S1_TOPS_Frame.csh, the terminal reports a bunch of errors mainly about some missing .grd files, such as r.grd and flt.grd, and I wonder why this would happen.

**OS and software version**
If applicable, what operating system and software version you are using?
OS: Ubuntu 20.04
GMTSAR version: v6.1

**Reproduce the problem**
![1](https://user-images.githubusercontent.com/46470946/155844689-503381b3-5801-4e85-90cd-a06952db26ba.png)
.SAFE files and orbit files
![2](https://user-images.githubusercontent.com/46470946/155844692-ed3df07c-76d3-4c5f-9a82-240584070567.png)
dem file
![3](https://user-images.githubusercontent.com/46470946/155844694-42eeb070-b99b-4c3b-833c-098aa4eb10f5.png)
process.sh

**Log file**
[log.txt](https://github.com/gmtsar/gmtsar/files/8146638/log.txt)

[Also if this is a general question that's not particlarly related to GMTSAR, such as "how to correct atmospheric noise" or "how to intepret my interferogram", consider using the discussions page]

**Comment by Xiaohua-Eric-Xu** [comment]
You misplaced the two SAFE files I think. Switch them and it'll run through.

**Comment by Uncleyyyyy** [comment]
> You misplaced the two SAFE files I think. Switch them and it'll run through.

Thanks for your reply! Now, the process of interferogram can run smoothly without reporting any error and the result of phase image is good. However, when unwrapping, there are many errors interupting the process. It tells that there are some other files missing. Could you please take another look at this problem? 

snaphu v2.0.1
32 parameters input from file snaphu.conf.brief (297 lines total)
Reading wrapped phase from file phase.in
No weight file specified.  Assuming uniform weights
Reading correlation data from file corr.in
Calculating deformation-mode cost parameters
Building range cost arrays
Building azimuth cost arrays
Initializing flows with MST algorithm
Killed
xyz2grd [ERROR]: Cannot find file unwrap.out
xyz2grd [ERROR]: Cannot find file conncomp.out
grdmath [ERROR]: tmp.grd is not a number, operator or file name
mv: 无法获取'tmp.grd' 的文件状态(stat): 没有那个文件或目录
grdmath [INFORMATION]: Enable all available threads (up to 4)
grdmath [INFORMATION]: Perform reverse Polish notation calculations on grids
grdmath [ERROR]: unwrap.grd is not a number, operator or file name
mv: 无法获取'tmp.grd' 的文件状态(stat): 没有那个文件或目录
grdgradient [ERROR]: Cannot find file unwrap.grd
grdinfo [ERROR]: Cannot find file unwrap.grd
grdinfo [ERROR]: Must specify one or more input files
makecpt [ERROR]: Option T: Must specify valid min[/max[/inc[<unit>|+n]]] option
grdinfo [ERROR]: Cannot find file unwrap.grd
grdinfo [ERROR]: Must specify one or more input files
grdinfo [ERROR]: Cannot find file unwrap.grd
grdinfo [ERROR]: Must specify one or more input files
grdimage [ERROR]: Cannot find file unwrap.grd
grdimage [ERROR]: Option -I: Must specify intensity file, value, or modifiers
psscale [ERROR]: Could not parse nwrap.grd into Geographical, Cartesian, or Temporal coordinates!
psscale [ERROR]: Option -R parsing failure. Correct syntax:
	-R<xmin>/<xmax>/<ymin>/<ymax>[/<zmin>/<zmax>]
	  Append +r if giving lower left and upper right coordinates
	-Rg or -Rd for global domain
	-R<grdfile> to take the domain from a grid file
psscale [ERROR]: Offending option -Runwrap.grd
psconvert [ERROR]: The file unwrap.ps has no BoundingBox in the first 20 lines or last 256 bytes. Use -A option.
Unwrapped phase map: unwrap.pdf

**Comment by Xiaohua-Eric-Xu** [comment]
The log file says it got `Killed`. My suspicion is your system has not enough memory to handle the snaphu program.

**Comment by Uncleyyyyy** [comment]
> The log file says it got `Killed`. My suspicion is your system has not enough memory to handle the snaphu program.

Thanks for your reply! Now, the process of interferogram can run smoothly without reporting any error and the outcome is good. However, when unwrapping, it will still report that some unwrap files are missing. Is it caused by my low memory also? By the way, I have set my RAM of virtual machine to 12G.

Setting search radius to be 300 ...
Reading in original grid...
Interpolating to nearest neighbour...
Working on line 8783 ...
  -1736116871 number of searches used ...
WRITING GRID IMAGE: Width x Heihgt = 17099 x 8783...
unwrapping phase with snaphu - higher threshold for faster unwrapping 

snaphu v2.0.1
32 parameters input from file snaphu.conf.brief (297 lines total)
Reading wrapped phase from file phase.in
No weight file specified.  Assuming uniform weights
Reading correlation data from file corr.in
Calculating deformation-mode cost parameters
Building range cost arrays
Building azimuth cost arrays
Initializing flows with MST algorithm
Killed
xyz2grd [ERROR]: Cannot find file unwrap.out
xyz2grd [ERROR]: Cannot find file conncomp.out
grdmath [ERROR]: tmp.grd is not a number, operator or file name
mv: 无法获取'tmp.grd' 的文件状态(stat): 没有那个文件或目录
grdmath [INFORMATION]: Enable all available threads (up to 4)
grdmath [INFORMATION]: Perform reverse Polish notation calculations on grids
grdmath [ERROR]: unwrap.grd is not a number, operator or file name
mv: 无法获取'tmp.grd' 的文件状态(stat): 没有那个文件或目录
grdgradient [ERROR]: Cannot find file unwrap.grd
grdinfo [ERROR]: Cannot find file unwrap.grd
grdinfo [ERROR]: Must specify one or more input files
makecpt [ERROR]: Option T: Must specify valid min[/max[/inc[<unit>|+n]]] option
grdinfo [ERROR]: Cannot find file unwrap.grd
grdinfo [ERROR]: Must specify one or more input files
grdinfo [ERROR]: Cannot find file unwrap.grd
grdinfo [ERROR]: Must specify one or more input files
grdimage [ERROR]: Cannot find file unwrap.grd
grdimage [ERROR]: Option -I: Must specify intensity file, value, or modifiers
psscale [ERROR]: Could not parse nwrap.grd into Geographical, Cartesian, or Temporal coordinates!
psscale [ERROR]: Option -R parsing failure. Correct syntax:
	-R<xmin>/<xmax>/<ymin>/<ymax>[/<zmin>/<zmax>]
	  Append +r if giving lower left and upper right coordinates
	-Rg or -Rd for global domain
	-R<grdfile> to take the domain from a grid file
psscale [ERROR]: Offending option -Runwrap.grd
psconvert [ERROR]: The file unwrap.ps has no BoundingBox in the first 20 lines or last 256 bytes. Use -A option.
Unwrapped phase map: unwrap.pdf

**Comment by Xiaohua-Eric-Xu** [comment]
17099 x 8783 is a very large grid that may require more than 12GB memory for snaphu. A typical 6000*6000 requires about 8GB. My suggestion is to either downsample or cut your grid before unwrapping, or expand your RAM.

**Comment by insar-dev** [comment]
snaphu can process large grid through spliting grid to tiles.
![image](https://user-images.githubusercontent.com/1676595/156497196-0bbe2a2f-34f2-4b29-b866-2d884faa12a1.png)

**Comment by Xiaohua-Eric-Xu** [comment]
@insar-dev have you tested this option before and potentially the -nproc option? Do they generate exactly the same solution as single, no-tile process?

**Comment by insar-dev** [comment]
snaphu:2.0.5
I have tested some data,  i think it is ok.
In tile mode, it is better than gamma.
snaphu is too too too too too too too too too too slow.

**Comment by SteffanDavies** [comment]
I find it hard to predict snaphu memory usage. I'm currently reserving 8gb for each thread but sometimes it's not sufficient for large grids, so now I allocate 12-16gb for each thread on every dataset. Maybe later I can optimize this by grid size, however the tiling might be an interesting solution. 

However @Xiaohua-Eric-Xu the results should be tested against full unwrapping first I find it hard to believe they will generate the same solution in every case but I may be wrong...

**Comment by Xiaohua-Eric-Xu** [comment]
Yes, snaphu costs lots of memory. The unwrapping should be quite similar over areas with good quality. But when the interferogram is filled with residues, I am not sure how snaphu keeps their topology consistent over multiple tiles.
